internal/auth/authn/k8s.go

```diff
@@ -2,32 +2,52 @@ package authn
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
+	"io"
 	"net/http"
-	"strings"
 )
 
-type K8sAuthN struct{}
+type K8sAuthN struct {
+	K8sApiUrl string
+}
 
-type ctxKeyAuthHeader string
+type OauthServerResponse struct {
+	TokenEndpoint string `json:"token_endpoint"`
+}
 
-const K8sTokenKey ctxKeyAuthHeader = "k8s-token"
+func (k8s K8sAuthN) ValidateToken(ctx context.Context, token string) (bool, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf("%s/apis/user.openshift.io/v1/users/~", k8s.K8sApiUrl), nil)
```

Comment:
This would be coupling very much to openshift.

Should we call this authn/openshift.go ? :)